Extension/src/main.ts

```diff
@@ -130,3 +133,37 @@ function sendTelemetry(info: PlatformInformation): void {
     telemetryProperties['infoArchitecture'] = info.architecture;
     Telemetry.logDebuggerEvent("acquisition", telemetryProperties);
 }
+
+export function HandleInsidersPrompt(): void {
+    const targetPopulation: TargetPopulation = util.getTargetPopulation();
+    // Only display updateChannel deprecated message if not running an Insiders build.
+    if (targetPopulation === TargetPopulation.Public) {
+        const settings: CppSettings = new CppSettings();
+        const displayedInsidersPrompt: PersistentState<boolean> = new PersistentState<boolean>("CPP.displayedInsidersPrompt", false);
+        // Only display updateChannel deprecated message if updateChannel is set to "Insiders".
+        if (settings.updateChannel === "Insiders") {
+            if (!displayedInsidersPrompt.Value) {
+                displayedInsidersPrompt.Value = true;
+                const message: string = localize('updateChannel.changed', "The `C_Cpp.updateChannel` setting is deprecated. Do you want to enable install of pre-releases of the C/C++ extension via the Marketplace?");
+                const yes: string = localize("yes.button", "Yes");
+                const no: string = localize("no.button", "No");
+                vscode.window.showInformationMessage(message, yes, no).then((selection) => {
+                    switch (selection) {
+                        case yes:
+                            vscode.commands.executeCommand("workbench.extensions.installExtension", "ms-vscode.cpptools", { installPreReleaseVersion: true });
```

Comment:
What happens when there isn't a pre-release version available (e.g. when we release 1.8.0)?  Does this just sign them up for insiders?

Reply:
That is my understanding.  But, we should ask isidorn , just in case.

@isidorn , Can you comment on the behavior of installing with `installPreReleaseVersion` set to true?  If there is no pre-release version available, or if the user is already using a more recent (non-pre-)release version, will it remain on the current version of the extension (and update to a pre-release when it becomes available) ?

Reply:
Passing `installPreReleaseVersion` flag will install the latest version, it can be a pre-release version or a release version, which ever is greater. If it happens that the latest version is a pre-release version then the user gets signed up to pre-release versions and gets auto update to pre-releases. Otherwise not, user gets auto updated only to release versions. In summary, this is not a flag that will sign up users to pre-releases. Hope it is clear.

Reply:
@sandy081 @isidorn .  It sounds like you understand what we are trying to do.  Our Insiders release channel is a very important part of the process of validating our extension.  However, it's a significant inconvenience to have to maintain an otherwise unnecessary prerelease version to ensure users are properly enrolled in receiving pre-release updates.  We may need to open a feature request to address this, and it may delay us from adopting platform-specific VSIX's.

Reply:
@Colengms I think your request makes sense and I will add that feature soon - https://github.com/microsoft/vscode/issues/139528